### PR TITLE
The End of Entity (Komm, Süsser Todd: Part 1.0)

### PR DESCRIPTION
**Pull request by gradientvera**
~~Merge https://github.com/space-wizards/RobustToolbox/pull/2296 first.~~
Required by https://github.com/space-wizards/space-station-14/pull/5624

- Makes `IEntity` a sealed class instead of an interface.
- Removes `Entity`.
- Makes all `Entity` usages use `IEntity` instead.

See commits.
I won't change `IEntity` to be called `Entity` because it'd be a massive, breaking refactor for a type we're removing soon anyway.

"It all returns to nothing...
It all comes tumbling down,
tumbling down, tumbling down"

### CODE REVIEW

**Comment by metalgearsloth** [comment]
https://www.youtube.com/watch?v=6kguaGI7aZg

**Comment by gradientvera** [comment]
(tests failing unrelated to PR, blame PaulVS instead)

File: Robust.Shared/GameObjects/IEntity.cs [hunk 1]
```diff
@@ -5,200 +5,205 @@
...
+        public bool HasComponent(Type type)
+        {
+            return EntityManager.HasComponent(Uid, type);
+        }
+
+        public T GetComponent<T>()
+        {
+            DebugTools.Assert(!Deleted, "Tried to get component on a deleted entity.");
+
+            return (T)EntityManager.GetComponent(Uid, typeof(T));
```
Comment (PaulRitter) [inline Robust.Shared/GameObjects/IEntity.cs]: why not
```suggestion
            return EntityManager.GetComponent<T>(Uid);
```
?

**Review by PaulRitter — CHANGES_REQUESTED** [review]

**Comment by PaulRitter** [comment]
i reran tests bc master tests dont fail anymore

File: Robust.Shared/GameObjects/IEntity.cs (hunk 1 above)
Comment (gradientvera) [reply Robust.Shared/GameObjects/IEntity.cs]: doesn't matter either way, oldcode

**Comment by gradientvera** [comment]
> i reran tests bc master tests dont fail anymore

a bit useless if master isn't merged to this branch...

**Comment by PaulRitter** [comment]
> > i reran tests bc master tests dont fail anymore
> 
> a bit useless if master isn't merged to this branch...

good point good point

**Comment by PaulRitter** [comment]
tests failing for real now

**Comment by gradientvera** [comment]
> tests failing for real now

Paul, I'm gonna fucking remove your certified git user card.

**Comment by PaulRitter** [comment]
> > tests failing for real now
> 
> Paul, I'm gonna fucking remove your certified git user card.

excuse?

**Comment by gradientvera** [comment]
> > > tests failing for real now
> > 
> > 
> > Paul, I'm gonna fucking remove your certified git user card.
> 
> excuse?

cause you still hadn't merged master into this branch lol

**Comment by PaulRitter** [comment]
> > > > tests failing for real now
> > > 
> > > 
> > > Paul, I'm gonna fucking remove your certified git user card.
> > 
> > 
> > excuse?
> 
> cause you still hadn't merged master into this branch lol

grafik

**Pull request merged**